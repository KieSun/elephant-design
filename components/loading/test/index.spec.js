import React from 'react'
import { mount, render } from 'enzyme'
import Loading from '../index'

describe('Loading', () => {
  it('affect the element when set props', () => {
    const wrapper = mount(
      <Loading text="Loading" indicatorColor="red" vertical size={20} />
    )
    const indicatorStyle = wrapper
      .find('.eleph-loading__indicator')
      .prop('style')
    expect(wrapper.find('.eleph-loading__text').text()).toBe('Loading')
    expect(indicatorStyle.borderColor).toBe('red')
    expect(indicatorStyle.width).toBe('20px')
  })

  it("should render custom indicator when it's set", () => {
    const customIndicator = <div className="custom-indicator" />
    const wrapper = render(
      <Loading indicator={customIndicator} text="Loading" />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
