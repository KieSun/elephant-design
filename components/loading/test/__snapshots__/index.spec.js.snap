// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Loading should render custom indicator when it's set 1`] = `
<div
  class="eleph-loading"
>
  <div
    class="custom-indicator"
  />
  <span
    class="eleph-loading__text"
  >
    Loading
  </span>
</div>
`;
